gridstatus/ercot_api/ercot_api.py

```diff
@@ -1348,23 +1348,32 @@ def get_system_load_charging_4_seconds(
             **api_params,
         )
 
-        data["AGCExecTimeUTC"] = pd.to_datetime(
-            data["AGCExecTimeUTC"],
-            utc=True,
-        ).dt.tz_convert(
+        # During the DST end transition, ERCOT publishes duplicate values like this:
+        # AGCExecTime         DSTFlag AGCExecTimeUTC       SystemDemand ESRChargingMW
+        # 2025-11-02T01:13:33 False   2025-11-02T07:13:33  42476.2188   2192.14233
+        # 2025-11-02T01:13:33 True    2025-11-02T07:13:33  41154.0469   1275.81592
+        # 2025-11-02T01:13:33 False   2025-11-02T06:13:33  42476.2188   2192.14233
+        # Even though the UTC times are different between rows 1 and 3, they have the
+        # same system demand and charging values and must be duplicates because the odds
+        # of these values being the same one hour apart is infinitesimal. So, we drop
+        # one of rows 1 and 3 and don't use the UTC time.
+        data = data.drop_duplicates(subset=["AGCExecTime", "DSTFlag", "SystemDemand"])
```

Comment:
The one hesitation I have is that clearly something is broken with this specific hour of data of data.

We are now adding a drop duplicates call that will hundred thousand+ times through the year to solve a problem that occurs on a single hour. 

It not only seems wildly inefficient, but also feels like it adds risk to solve a very isolated problem.

Yep ideas come to mind

1. do a one off fill of this hour right now and then leave the code as-is.

2. Only run the drop duplicates code when requested date range contains a dst change

Reply:
> To test how much overhead the current solution adds, I ran

also, you analysis doesn't look at memory, which is more often our bottleneck that causes server crashes.

still, i dont think this is about performance. the goal is to just make this code as safe as possible. I just always worry about unintended consequences of `drop_duplicates`

Reply:
I'm with you on avoiding unnecessary `drop_duplicates`.